app/src/main/java/org/p2p/wallet/rpc/RpcModule.kt

```diff
@@ -3,48 +3,128 @@ package org.p2p.wallet.rpc
+import org.p2p.wallet.rpc.repository.amount.RpcAmountRemoteRepository
+import org.p2p.wallet.rpc.repository.amount.RpcAmountRepository
+import org.p2p.wallet.rpc.repository.account.RpcAccountRemoteRepository
+import org.p2p.wallet.rpc.repository.account.RpcAccountRepository
+import org.p2p.wallet.rpc.repository.amount.RpcAmountInMemoryRepository
+import org.p2p.wallet.rpc.repository.amount.RpcAmountInteractor
+import org.p2p.wallet.rpc.repository.amount.RpcAmountLocalRepository
+import org.p2p.wallet.rpc.repository.balance.RpcBalanceInMemoryRepository
+import org.p2p.wallet.rpc.repository.balance.RpcBalanceInteractor
+import org.p2p.wallet.rpc.repository.balance.RpcBalanceLocalRepository
+import org.p2p.wallet.rpc.repository.balance.RpcBalanceRemoteRepository
+import org.p2p.wallet.rpc.repository.balance.RpcBalanceRepository
+import org.p2p.wallet.rpc.repository.blockhash.RpcBlockHashRemoteRepository
+import org.p2p.wallet.rpc.repository.blockhash.RpcBlockHashRepository
+import org.p2p.wallet.rpc.repository.signature.RpcSignatureRemoteRepository
+import org.p2p.wallet.rpc.repository.signature.RpcSignatureRepository
+import org.p2p.wallet.rpc.repository.token.RpcTokenRemoteRepository
+import org.p2p.wallet.rpc.repository.token.RpcTokenRepository
+import org.p2p.wallet.rpc.repository.transaction.RpcTransactionRemoteRepository
+import org.p2p.wallet.rpc.repository.transaction.RpcTransactionRepository
 
 object RpcModule : InjectionModule {
 
     override fun create() = module {
-        single {
-            val serverErrorInterceptor = ServerErrorInterceptor(get())
-            val serum = getRetrofit(Environment.SOLANA.endpoint, interceptor = serverErrorInterceptor)
-            val serumRpcApi = serum.create(RpcApi::class.java)
-
-            val mainnet = getRetrofit(Environment.MAINNET.endpoint, interceptor = serverErrorInterceptor)
-            val mainnetRpcApi = mainnet.create(RpcApi::class.java)
-
-            /* This string is in gitignore and it's null when CI/CD runs some actions */
-            val rpcPoolApiKey = BuildConfig.rpcPoolApiKey
-            val baseUrl = if (rpcPoolApiKey.isNotBlank()) {
-                "${Environment.RPC_POOL.endpoint}$rpcPoolApiKey/"
-            } else {
-                Environment.RPC_POOL.endpoint
-            }
-            val rpcpool = getRetrofit(baseUrl, interceptor = serverErrorInterceptor)
-            val rpcpoolRpcApi = rpcpool.create(RpcApi::class.java)
-
-            val testnet = getRetrofit(Environment.DEVNET.endpoint, interceptor = serverErrorInterceptor)
-            val testnetRpcApi = testnet.create(RpcApi::class.java)
-
-            RpcRemoteRepository(serumRpcApi, mainnetRpcApi, rpcpoolRpcApi, testnetRpcApi, get())
-        } bind RpcRepository::class
-
-        factory { CloseInteractor(get(), get()) }
-        factory { TransactionInteractor(get(), get(), get()) }
-
-        single { RpcAmountRemoteRepository(get()) } bind RpcAmountRepository::class
+        // Rpc account api
```

Comment:
you can remove need of the comments with just extracting method and naming it good